api/rest/client/methods.go

```diff
@@ -302,3 +304,46 @@ func statusReached(target api.TrackerStatus, gblPinInfo api.GlobalPinInfo) (bool
 	}
 	return true, nil
 }
+
+// AddMultiFile adds new files to the cluster, importing and potentially
+// sharding underlying dags across the ipfs daemons of multiple cluster peers.
+// Progress can be tracked by passing a channel onto which deliver AddedOutput
+// updates.
+func (c *Client) AddMultiFile(
```

Comment:
Big problem here: no function in the client takes a context. Requests cannot be cancelled. This will need to get done on a separate PR as it's pretty self-contained. What was I thinking?